File: julia/test_SST.jl
using Test
push!(LOAD_PATH, "../julia/")
using SST

y, m = SST.get_year_month_file("/data/SST/Global/monthly/2000/TERRA_MODIS.20000201_20000229.L3m.MO.SST4.sst4.9km.nc")
@test y == 2000
@test m == 2

sstclimfile = SST.get_monthly_clim_filename("TERRA", "MODIS", 2000, 2020, 2)
@test sstclimfile == "TERRA_MODIS.20000201_20200229.L3m.MC.SST4.sst4.9km.nc"
sstclimfile = SST.get_monthly_clim_filename("TERRA", "MODIS", 2000, 2020, 2, res="4km")
@test sstclimfile == "TERRA_MODIS.20000201_20200229.L3m.MC.SST4.sst4.4km.nc"

testfile = "/data/SST/Global/monthly/2000/TERRA_MODIS.20000201_20000229.L3m.MO.SST4.sst4.9km.nc"
@time lon, lat, sst = SST.read_sst_oceancolor_L3(testfile, [-20, -8., 25., 33.]);
@test length(lon) == 144
@test length(lat) == 96
@test lon[1] == -19.958334f0
@test lat[2] == 32.874996f0
@test sst[20, 30] == 18.15f0

File: tests/test_sst.py
import os
import glob
import datetime
import numpy as np
from filament import filament
import unittest
import logging


class TestSST(unittest.TestCase):

    def setUp(self):
        self.fname = "./data/TERRA_MODIS.20191201_20191231.L3m.MO.SST4.sst4.9km.nc"
        self.assertTrue(os.path.isfile(self.fname))

    def test_read_L3(self):
        sstmonth = filament.SST()
        sstmonth.read_from_oceancolorL3(self.fname)

        self.assertEqual(len(sstmonth.lat), 2160)
        self.assertEqual(len(sstmonth.lon), 4320)
        self.assertAlmostEqual(sstmonth.lon[0], -179.95833, places=5)
        self.assertEqual(sstmonth.lat[10], 89.125)
        self.assertTrue(np.ma.is_masked(sstmonth.field[200, 300]))
        self.assertAlmostEqual(sstmonth.field[1200, 300], 28.994999, places=6)


    def test_get_filename(self):
        sstfilename = filament.get_monthly_filename("TERRA", "MODIS", 2020, 6)
        self.assertTrue(sstfilename == "TERRA_MODIS.20200601_20200630.L3m.MO.SST4.sst4.9km.nc")

        sstfilename = filament.get_monthly_filename("TERRA", "MODIS", 2020, 6, "4km")
        self.assertTrue(sstfilename == "TERRA_MODIS.20200601_20200630.L3m.MO.SST4.sst4.4km.nc")

    def test_get_monthly_filename(self):
        sstfilename = filament.get_monthly_clim_filename("TERRA", "MODIS", 2000, 2019, 2, res="9km")
        self.assertTrue(sstfilename == "TERRA_MODIS.20000201_20190228.L3m.MC.SST4.sst4.9km.nc")

        sstfilename = filament.get_monthly_clim_filename("TERRA", "MODIS", 2000, 2019, 2, "4km")
        self.assertTrue(sstfilename == "TERRA_MODIS.20000201_20190228.L3m.MC.SST4.sst4.9km.nc")


if __name__ == '__main__':
    unittest.main()
